examples/api/lib/cupertino/modal_bottom_sheet/modal_bottom_sheet.0.dart

```diff
@@ -0,0 +1,373 @@
+  /// Creates a page route for use in an iOS designed app.
+  MBSPageRoute({
+    required this.pageBuilder,
+    required BuildContext context,
+  }) : super(
+    delegatedTransition: (Navigator.of(context).widget is MBSNavigator) ?
+      MBSTransition.secondaryDelegateTransition : MBSTransition.delegateTransition,
+  );
+
+  final WidgetBuilder pageBuilder;
+
+  @override
+  Widget buildContent(BuildContext context) => pageBuilder(context);
+
+  @override
+  Color? get barrierColor => const Color(0x20000000);
+
+  @override
+  bool get barrierDismissible => false;
+
+  @override
+  String? get barrierLabel => 'Stacked card appearance for modal bottom sheet';
+
+  @override
+  bool get maintainState => true;
+
+  @override
+  bool get opaque => false;
+}
+
+mixin MBSRouteTransitionMixin<T> on PageRoute<T> {
+  /// Builds the primary contents of the route.
+  @protected
+  Widget buildContent(BuildContext context);
+
+  @override
+  // A relatively rigorous eyeball estimation.
+  Duration get transitionDuration => const Duration(milliseconds: 500);
+
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+    final Widget child = buildContent(context);
+    return MBSNavigator(
+      parentNavigatorContext: context,
+      initialRoute: '/',
+      onGenerateRoute: (RouteSettings settings) {
+        return CupertinoPageRoute(
+          builder: (BuildContext context) {
+            return Semantics(
+              scopesRoute: true,
+              explicitChildNodes: true,
+              child: child,
+            );
+          }
+        );
+      }
+    );
+  }
+
+  /// Returns a [CupertinoFullscreenDialogTransition] if [route] is a full
+  /// screen dialog, otherwise a [CupertinoPageTransition] is returned.
+  ///
+  /// Used by [CupertinoPageRoute.buildTransitions].
+  ///
+  /// This method can be applied to any [PageRoute], not just
+  /// [CupertinoPageRoute]. It's typically used to provide a Cupertino style
+  /// horizontal transition for material widgets when the target platform
+  /// is [TargetPlatform.iOS].
+  ///
+  /// See also:
+  ///
+  ///  * [CupertinoPageTransitionsBuilder], which uses this method to define a
+  ///    [PageTransitionsBuilder] for the [PageTransitionsTheme].
+  static Widget buildPageTransitions<T>(
+    ModalRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return MBSTransition(
+      primaryRouteAnimation: animation,
+      secondaryRouteAnimation: secondaryAnimation,
+      child: child,
+    );
+  }
+
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+    return buildPageTransitions<T>(this, context, animation, secondaryAnimation, child);
+  }
+}
+
+class MBSNavigator extends Navigator {
```

Comment:
Is the intention that this would be brought into the framework, or would a user that wants to define their own transition (in the same way that MBS will be defined) have to extend Navigator too? I guess the question is whether or not a custom pop behavior (popMBS here) is exactly the behavior we want to enable for everyone defining their own transition.